src/System.Numerics.Vectors/tests/CommonHashCodeTests.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.Numerics.Tests
+{
+    public static class CommonHashCodeTests
+    {
+        private const int ElementsCount = 100000;
+
+        /// <summary>
+        /// Initial version of hash function in System.Numerics.Vectors.
+        /// </summary>
+        private static int BaseHashCodeCombine(int h1, int h2)
+        {
+            return ((h1 << 5) + h1) ^ h2;
+        }
+
+        private static void CountVector2Collisions(Func<int, Vector2> vectorConstructor, out int baseCollisions,
+            out int collisions)
+        {
+            baseCollisions = Enumerable.Range(0, ElementsCount)
+                .Select(vectorConstructor)
+                .GroupBy(x => BaseHashCodeCombine(x.X.GetHashCode(), x.Y.GetHashCode()))
+                .Where(t => t.Count() > 1)
+                .Select(t => t.ToList().Count)
+                .Sum();
+
+            collisions = Enumerable.Range(0, ElementsCount)
+                .Select(vectorConstructor)
+                .GroupBy(x => x.GetHashCode())
+                .Where(t => t.Count() > 1)
+                .Select(t => t.ToList().Count)
+                .Sum();
+        }
+
+        private static void CountVectorCollisions<T>(Func<int, Vector<T>> vectorConstructor, out int baseCollisions,
+            out int collisions) where T : struct
+        {
+            baseCollisions = Enumerable.Range(0, ElementsCount)
+                .Select(vectorConstructor).
+                GroupBy(x =>
+                {
+                    int hash = 1;
+                    for (int i = 0; i < Vector<T>.Count; i++)
+                    {
+                        hash = BaseHashCodeCombine(hash, x[i].GetHashCode());
+                    }
+                    return hash;
+                })
+                .Where(t => t.Count() > 1)
+                .Select(t => t.ToList().Count)
+                .Sum();
+
+            collisions = Enumerable.Range(0, ElementsCount)
+                .Select(vectorConstructor)
+                .GroupBy(x => x.GetHashCode())
+                .Where(t => t.Count() > 1)
+                .Select(t => t.ToList().Count)
+                .Sum();
+        }
+
+        [Fact]
+        public static void CombineHashCodesVector2Tests()
```

Comment:
1. Why are we using a test that compares the collisions between old and new versions?  What happens if somebody wants to change the hashing to something else - do they need to keep all three versions?  What if the third version is designed to be weaker (have more collisions) than this one?  Wouldn't it be better to instead just have the test check for the percentage of collisions vs total elements?  I mean, if the current method has a 100% collision rate and the new one a 99.99999% rate, do we want either?  The design decision being made isn't "we need a hashing method with lower collisions", it's "we need a hashing method with collisions below this level (and by implication, our current method fails that test)".  The comparison, new vs old, is meta information (like performance tests, or approximation amount in various calculations), probably not something that should be encoded as a test.

 2. Collision tests should be exhaustive, and are likely to require `OuterLoop`.  You probably also want tests for:
   - One element `0`, the other an incrementing value (negative or positive).
   - `X` much larger than `Y` (ie, flip the elements of your current first test)
   - Identical elements.
   - Negative elements.
   - Elements at the extreme of the range (max/min).
   - Elements differing by the prime and nearest power-of-2 (`32`, here).  

 Note that you should be putting the "lead" element in both slots for a two-element vector (since you might be ignoring either one).  Given the general-use nature of `Vector`, tests checking for the percentage of collisions should be doing it as the aggregate of all categories (as opposed to a specific implementation which might have lower collisions for positive values or something).
 
 3. Your "not greater than 10" test won't have more than 10 different elements in it, and you generate each one 10,000 times.  What results are you expecting to get there?  Note that for general-purpose use the number of collisions over such a miniscule (although common) range are liable to be meaningless.  What happens to this test if the current number of collisions is 0 and the rest of the range is near 100%, but the new method puts 1 in 0-10 and only 20% elsewhere?  Note that a good property of hashes is the "randomness" of the distribution, so this isn't an idle concern.
 4. Speaking of distribution, that might be another good test to add here.